File: 0905-Sort-Array-By-Parity.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParity = function (nums) {
    return [...nums.filter((_) => _ % 2 == 0), ...nums.filter((_) => _ % 2 == 1)];
};

/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParity = function (nums) {
    return nums.reduce((acc, item) => {
        if (item % 2 == 0) acc.unshift(item);
        else acc.push(item);
        return acc;
    }, []);
};

File: 0905-Sort-Array-By-Parity.go
func sortArrayByParity(nums []int) []int {
	ans := make([]int, 0)
	for _, e := range nums {
		if e%2 == 0 {
			ans = append([]int{e}, ans...)
		} else {
			ans = append(ans, e)
		}
	}
	return ans
}